docs/user/registering-microshift-devices-acm.md

```diff
@@ -0,0 +1,157 @@
+      generation: 1
+      labels:
+        fleet: acm-test
+    spec:
+      os:
+        image: quay.io/myorg/device-image-with-microshift:v1
+      config:
+      - configType: HttpConfigProviderSpec
+        httpRef:
+          filePath: /var/local/crd.yaml
+          repository: acm-test
+          suffix: /agent-registration/crds/v1
+        name: acm-crd
+      - configType: HttpConfigProviderSpec
+        httpRef:
+          filePath: /var/local/import.yaml
+          repository: acm-test
+          suffix: /agent-registration/manifests/{{ device.metadata.name }}
+        name: acm-import
+      hooks:
+        afterUpdating:
+          - path: "/var/local/crd.yaml"
+            onFile: [Create]
+            actions:
+              - executable:
+                  run: "kubectl apply -f /var/local/crd.yaml"
+                  envVars: ["KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig"]
+          - path: "/var/local/import.yaml"
+            onFile: [Create]
+            actions:
+              - executable:
+                  run: "kubectl apply -f /var/local/import.yaml"
+                  envVars: ["KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig"]
+```
+
+### Fleet Specification Breakdown
+
+As described in the user documentation, a fleet specification is composed of various sections. Let us deep dive in the config and hooks section of our sample fleet.
+
+1. **Configuration**: The configuration section uses the HTTP configuration provider to fetch information from an endpoint. The repository `acm-test` contains the registration URL for ACM's agent registration:
+
+    ```console
+    - configType: HttpConfigProviderSpec
+      httpRef:
+        filePath: /var/local/crd.yaml
+        repository: acm-test
+        suffix: /agent-registration/crds/v1
+      name: acm-crd
+    ```
+
+    This retrieves the CRD from the ACM endpoint and stores it at the specified file path.
+
+    Basically, it is like making an API call against the endpoint:
+
+    `https://$agent_registration_host/agent-registration/crds/v1`
+
+    The next configuration retrieves the cluster import manifests. As shown below, the HttpConfigProviderSpec supports Flight Control template mechanism, so the device name can be used as part of the suffix.
+
+    ```console
+    - configType: HttpConfigProviderSpec
+      httpRef:
+        filePath: /var/local/import.yaml
+        repository: acm-test
+        suffix: /agent-registration/manifests/{{ device.metadata.name }}
+      name: acm-import
+    ```
+
+    This API call to ACM's agent registration endpoint will retrieve a set of Kubernetes manifests to deploy the Klusterlet. Once we have both the Klusterlet CRD and deployment manifests, we will use Flight Control hooks to apply them.
+
+1. **Hooks**: Once the configuration is fetched, the hooks apply the Kubernetes manifests to the device:
+
+```console
+hooks:
+  afterUpdating:
+    - path: "/var/local/crd.yaml"
+      onFile: [Create]
```

Comment:
Hooks might change, so we will have to update this documentation, but in the meantime, this works. 

MicroShift has the capability to apply manifests at a certain location `/etc/microshift/manifests`, and this will happen everytime Microshift systemd service starts (so at every reboot). Once Klusterlet agent is started in the MicroShift cluster, ACM will be in charge for managing it. This is why this solution didn't work. 

The reboot policy would have the same behavior as the MicroShift applier, so I didn't want to use that, because it would bring up the same questions.

In the demo I'm recording, I basically run devices with a bootstrap image with flightctl-agent and microshift, so image is already reconciled... that would allow hooks to work because microshift binary is already present in the device.